src/main/java/bio/terra/service/auth/iam/sam/SamApiService.java

```diff
@@ -0,0 +1,73 @@
+package bio.terra.service.auth.iam.sam;
+
+import bio.terra.app.configuration.SamConfiguration;
+import bio.terra.service.configuration.ConfigEnum;
+import bio.terra.service.configuration.ConfigurationService;
+import java.util.concurrent.TimeUnit;
+import org.broadinstitute.dsde.workbench.client.sam.ApiClient;
+import org.broadinstitute.dsde.workbench.client.sam.api.GoogleApi;
+import org.broadinstitute.dsde.workbench.client.sam.api.GroupApi;
+import org.broadinstitute.dsde.workbench.client.sam.api.ResourcesApi;
+import org.broadinstitute.dsde.workbench.client.sam.api.StatusApi;
+import org.broadinstitute.dsde.workbench.client.sam.api.TermsOfServiceApi;
+import org.broadinstitute.dsde.workbench.client.sam.api.UsersApi;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SamApiService {
+
+  private final SamConfiguration samConfig;
+  private final ConfigurationService configurationService;
+
+  @Autowired
+  public SamApiService(SamConfiguration samConfig, ConfigurationService configurationService) {
+    this.samConfig = samConfig;
+    this.configurationService = configurationService;
+  }
+
+  public ResourcesApi resourcesApi(String accessToken) {
+    return new ResourcesApi(createApiClient(accessToken));
+  }
+
+  public StatusApi statusApi() {
+    return new StatusApi(createUnauthApiClient());
+  }
+
+  public GoogleApi googleApi(String accessToken) {
+    return new GoogleApi(createApiClient(accessToken));
+  }
+
+  public UsersApi usersApi(String accessToken) {
+    return new UsersApi(createApiClient(accessToken));
+  }
+
+  public TermsOfServiceApi termsOfServiceApi(String accessToken) {
+    return new TermsOfServiceApi(createApiClient(accessToken));
+  }
+
+  public GroupApi groupApi(String accessToken) {
+    return new GroupApi(createApiClient(accessToken));
+  }
+
+  private ApiClient createUnauthApiClient() {
+    ApiClient apiClient = new ApiClient();
```

Comment:
I don't love that I'm constructing a new `ApiClient` here.  I think true unit tests would allow us to mock this creation and verify that we were passing the expected timeout overrides to the mock.  But this code was starting to feel like a set of matryoshka dolls :) anyways, I welcome other ideas.

Reply:
I think it's OK to create `ApiClient` here. Regarding creation, I don't think you can share API clients with different callers since I don't think they're thread-safe. I don't think it's worth making another layer of mocks for this one class. The only test that doesn't use a mocked `SamApiService` is `SamApiServiceTest` itself anyway. If you did use `spy` and override this code in a mock, then you wouldn't get coverage for these lines.

Reply:
Great, thanks for the the thoughts! I feel more comfortable leaving this as-is in that case.